main.go

```diff
@@ -381,20 +421,50 @@ func runCLIApp(c *cli.Context) (err error) {
 	{
 		mfs, err = mountWithArgs(bucketName, mountPoint, flags, mountConfig)
 
-		if err == nil {
+		// This utility is to absorb the error
+		// returned by daemonize.SignalOutcome calls by simply
+		// logging them as error logs.
+		callDaemonizeSignalOutcome := func(err error) {
+			if err2 := daemonize.SignalOutcome(err); err2 != nil {
+				logger.Errorf("Failed to signal error to parent-process from daemon: %v", err2)
+			}
+		}
+
+		markSuccessfulMount := func() {
 			// Print the success message in the log-file/stdout depending on what the logger is set to.
 			logger.Info(SuccessfulMountMessage)
+			callDaemonizeSignalOutcome(nil)
+		}
 
-			daemonize.SignalOutcome(nil)
-		} else {
+		markMountFailure := func(err error) {
 			// Printing via mountStatus will have duplicate logs on the console while
 			// mounting gcsfuse in foreground mode. But this is important to avoid
 			// losing error logs when run in the background mode.
 			logger.Errorf("%s: %v\n", UnsuccessfulMountMessagePrefix, err)
 			err = fmt.Errorf("%s: mountWithArgs: %w", UnsuccessfulMountMessagePrefix, err)
-			daemonize.SignalOutcome(err)
-			return
+			callDaemonizeSignalOutcome(err)
+		}
+
+		if err != nil {
+			markMountFailure(err)
+			return err
+		}
+		if !isDynamicMount(bucketName) {
```

Comment:
Why are we calling the same thing in both background and foreground mode?
Only calling the recursive listing here should be suffice, right?

Reply:
> Only calling the recursive listing here should be suffice, right?

Right. No need to call it in the daemon's parent process. So, I've removed that bit  of code.